### PR TITLE
CI: Update install process for creating Docker images

### PR DESCRIPTION
**Pull request by sentree**
This mainly makes all of the changes necessary to create Docker images for each of the projects. The conditional logic in the install script has been converted to a more standardized format because it was annoying me and #8 has been addressed. I was getting a strange error on trying to build the catkin workspace in a Docker container, so this may also fix the elusive #7.

### CODE REVIEW

**Comment by kev-the-dev** [comment]
https://hub.docker.com/r/jenkinsci/pipeline-as-code-github-demo/

File: install.sh [hunk 1]
```diff
@@ -59,37 +59,26 @@ MILRC_FILE=$MIL_CONFIG_DIR/milrc
...
-	BVSDK_PASSWORD="`env | grep BVSDK_PASSWORD | grep -oe '[^=]*$'`"
-	ENABLE_USB_CAM=false
-	INSTALL_CUDA=true
+# Prompt the user to configure the installation if the script is not being used in Docker
+if [ "$DOCKER" != "true" ]; then
+
+	# Install no project by default, the user must select one
+	SELECTED="false"
+	INSTALL_SUB="false"
+	INSTALL_PRO="false"
```
Comment (DSsoto) [inline install.sh:68]: Is the pro version extra?

File: install.sh [hunk 2]
```diff
@@ -134,12 +123,12 @@ else
 				echo "Indigo, the SubjuGator repository at an earlier date,  and all of the old"
 				echo "SubjuGator dependencies will need to be downloaded and installed."
 				echo -n "Do you still wish to proceed? [y/N] " && read RESPONSE
-				if ([ "$RESPONSE" = "Y" ] || [ "$RESPONSE" = "y" ]); then
+				if ([ "$RESPONSE" == "Y" ] || [ "$RESPONSE" == "y" ]); then
```
Comment (DSsoto) [inline install.sh:126]: If these changes are purely cosmetic, then there is something _very_ wrong with bash.

File: user_install.sh [hunk 3]
```diff
@@ -60,10 +60,10 @@ MILRC_FILE=$MIL_CONFIG_DIR/milrc
 #======================#
 
 # Install no project by default, the user must select one
-SELECTED=false
-INSTALL_SUB=false
-INSTALL_PRO=false
-INSTALL_NAV=false
+SELECTED="false"
```
Comment (DSsoto) [inline user_install.sh:63]: I don't remember us having a separate user_install.sh. Can you explain this division?

**Review by DSsoto — APPROVED** [review]
I'm trusting that you've tested this. Will you be adding CI to the installer?

File: user_install.sh (hunk 3 above)
Comment (sentree) [reply user_install.sh:63]: See #3 for the answer to this. It mainly boils down to doing installs on SubjuGator and Shuttle, where the main script has already been run. The script generates, populates, and builds a user workspace without needing any root access. It was how I made all of your workspaces for the last pool day.

File: install.sh (hunk 2 above)
Comment (sentree) [reply install.sh:126]: The brackets are what defines a conditional statement in bash, not the operator. if you run `which [`, you will see that `/usr/bin/[` is actually a program belonging to the test package. This is part of POSIX shell and only supports `=` for a conditional check. In modern bash, this has been replaced with a `[[` built-in that supports both `==` and `=` for compatibility with POSIX. Realistically, I should be using `[[`, but bash allows me to shorthand it to `[`.

Comment (sentree) [reply install.sh:126]: I actually may change to `[[` for clarity, stay tuned.

File: install.sh (hunk 1 above)
Comment (sentree) [reply install.sh:68]: As always.

**Comment by sentree** [comment]
The CI images were built with this, so we know it works. I will be adding a feature to the CI that builds the Docker images with the script whenever changes are merged to the installer master branch.

**Pull request merged**